contracts/finance/VestingWalletConfidentialFactory.sol

```diff
@@ -0,0 +1,204 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import {FHE, euint64, externalEuint64, euint128} from "@fhevm/solidity/lib/FHE.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {Clones} from "@openzeppelin/contracts/proxy/Clones.sol";
+import {IConfidentialFungibleToken} from "./../interfaces/IConfidentialFungibleToken.sol";
+import {ERC7821WithExecutor} from "./ERC7821WithExecutor.sol";
+import {VestingWalletCliffConfidential} from "./VestingWalletCliffConfidential.sol";
+import {VestingWalletConfidential} from "./VestingWalletConfidential.sol";
+
+/**
+ * @dev This factory enables creating {VestingWalletCliffExecutorConfidential} in batch.
+ *
+ * Confidential vesting wallets created inherit both {VestingWalletCliffConfidential} for vesting cliffs
+ * and {ERC7821WithExecutor} to allow for arbitrary calls to be executed from the vesting wallet.
+ */
+contract VestingWalletConfidentialFactory {
+    address private immutable _vestingImplementation;
+
+    event VestingWalletConfidentialFunded(
+        address indexed vestingWalletConfidential,
+        address indexed beneficiary,
+        address indexed confidentialFungibleToken,
+        euint64 encryptedAmount,
+        uint48 startTimestamp,
+        uint48 durationSeconds,
+        uint48 cliffSeconds,
+        address executor
+    );
+    event VestingWalletConfidentialCreated(
+        address indexed vestingWalletConfidential,
+        address indexed beneficiary,
+        uint48 startTimestamp,
+        uint48 durationSeconds,
+        uint48 cliffSeconds,
+        address indexed executor
+    );
+
+    struct VestingPlan {
+        address beneficiary;
+        externalEuint64 encryptedAmount;
+        uint48 start;
```

Comment:
why is start part of the plan, but not duration ?

Reply:
That is not my experience. When cliff and duration are the same, start is also the same, particularly if they are part of a batch that was created at the same time.

I can understand the logic of "everybody in a batch is on the same plan". I can also understand the logic of "every plan is different" ... I don't understand the logic of "all plans share x and y, but z is individual".

Do we have details about what vesting plans _the user_ wants to create ?

Reply:
We have no details on what vesting plans will actually be created in batch.

But many companies create vesting plans for employees starting from different dates (depending on when they joined). I am also aware that the user plans to create vesting plans for employees.

Reply:
Brought the duration and cliff into the struct